File: bindings/kotlin/examples/GenerateMnemonic.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.MnemonicLength
import iota_sdk.generateMnemonic
import kotlin.io.println

fun main() {
    val mnemonic = generateMnemonic(null)
    println("24 word mnemonic: $mnemonic")
    val mnemonic12 = generateMnemonic(MnemonicLength.WORDS12)
    println("12 word mnemonic: $mnemonic12")
}

File: bindings/swift/examples/generate_mnemonic.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct GenerateMnemonicExample {
  static func main() throws {
    let mnemonic24 = generateMnemonic(wordCount: nil)
    print("24 word mnemonic:", mnemonic24)
    let mnemonic12 = generateMnemonic(wordCount: MnemonicLength.words12)
    print("12 word mnemonic:", mnemonic12)
  }
}
